be/src/service/backend_options.cpp

```diff
@@ -57,6 +62,10 @@ bool BackendOptions::init() {
             VLOG_CRITICAL << "check ip=" << addr_it->get_host_address_v4();
             if ((*addr_it).is_loopback_v4()) {
                 loopback = addr_it->get_host_address_v4();
+            } else if (!_s_net_interfaces.empty()) {
+                if (is_in_net_interface(addr_it->get_host_address_v4())) {
+                    _s_localhost = addr_it->get_host_address_v4();
+                }
```

Comment:
We should not follow cidrs here.

There is a problem here, e.g. There are 2 nics, eth0 and eth1, we configed eth1 and cidrs, if we tranverse eth0 first then cidr is used. we should let network interface is selected in priority. 

So we should handle  net_interface before this while.